adapters/iqzone/iqzone.go

```diff
@@ -23,19 +23,70 @@ func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters
 	return bidder, nil
 }
 
-func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
-	requestJSON, err := json.Marshal(request)
-	if err != nil {
-		return nil, []error{err}
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errs []error
+	var err error
+	var bidderExt adapters.ExtImpBidder
+	var iqzoneExt openrtb_ext.ImpExtIQZone
+	var adapterRequests []*adapters.RequestData
+
+	reqCopy := *request
+	for _, imp := range request.Imp {
+		reqCopy.Imp = []openrtb2.Imp{imp}
+
+		if err = json.Unmarshal(reqCopy.Imp[0].Ext, &bidderExt); err != nil {
```

Comment:
Declaring `bidderExt` and `iqzoneExt` at the start rather than in the loop, means they are persistent. This could potentially cause an issue if a field that exists in `imp[0]` is omitted from `imp[1]`, the Unmarshal call will not zero out that parameter and thus you could see it where it does not belong.

Reply:
Thanks for finding this bug and explaining it. We have fixed this.